### PR TITLE
Include list of merged PRs in dependency bumps for jupyterhub repos

### PR DESCRIPTION
**Pull request by manics**
This script takes a repository, current tag/commit/chart-version/docker-tag, new tag/commit/chart-version/docker-tag, and outputs a list of PRs in markdown that is used in a pull request body.

~Untested.... though I think it's ready to try.~

Closes https://github.com/jupyterhub/mybinder.org-deploy/issues/2290

### CODE REVIEW

File: scripts/get-prs.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+gh = github.Github(token)
+r = gh.get_repo(args.repo)
+
+start = extract_gitref(args.start)
+end = extract_gitref(args.end)
+
+prs = set()
+git_compare = r.compare(start, end)
+for c in git_compare.commits:
+    s = gh.search_issues("", type="pr", repo=args.repo, sha=c.sha)
```
Comment (manics) [inline scripts/get-prs.py:49]: This use of the search API to lookup PRs associated with a commit is based on henchbot https://github.com/henchbot/mybinder.org-upgrades/blob/d84971953cdc3efecca7c2bdd408de80835afc4e/src/mybinder-upgrades/henchbot.py#L254-L255

but the search API has a [lower rate limit](https://docs.github.com/en/rest/search#rate-limit) than other API calls and I keep hitting it during testing.

There's a newer API endpoint which doesn't have this reduced rate limit: https://docs.github.com/en/rest/commits/commits#list-pull-requests-associated-with-a-commit
so I'll try that instead.

**Comment by manics** [comment]
I've tested this on my fork by pushing an additional commit on a test branch to
- modify the repo/branch check so that watch-dependencies can run on my fork/branch
- downgraded some versions to test that PRs are opened

This is the diff I used to test: https://github.com/manics/mybinder.org-deploy/compare/dependencies-list-prs..dependencies-list-prs-test?expand=1

And these are the PRs it's opened: https://github.com/manics/mybinder.org-deploy/pulls/app%2Fgithub-actions
https://github.com/manics/mybinder.org-deploy/pull/29 and https://github.com/manics/mybinder.org-deploy/pull/28 are examples of the descriptions with a list of PRs.

**Review by consideRatio — APPROVED** [review]
LGTM!

**Comment by consideRatio** [comment]
Nice work @manics, the result is great: https://github.com/jupyterhub/mybinder.org-deploy/pull/2367#issue-1409354381

image

**Pull request merged**